scripts/refresh-submodules.sh

```diff
@@ -25,12 +25,27 @@ submodules=(
 for ent in "${@:-${submodules[@]}}"; do
     submodule=${ent%%:*}
     [ ${submodule} == ${ent} ] && branch="master" || branch=${ent#*:}
-    GIT_DIR="$submodule/.git" git pull --ff-only origin ${branch}
-    SUMMARY=$(git submodule summary $submodule)
-    if grep '^ *<' <<< "$SUMMARY"; then
+
+    pushd .
+    cd $submodule
+    git fetch origin ${branch}
+    if ! git merge-base --is-ancestor HEAD origin/master; then
         echo "Non fast-forward changes detected! Fire three red flares from your flare pistol."
         exit 1
     fi
+    # collect the summary
+    head=$(git rev-parse --short=8 HEAD)
+    master=$(git rev-parse --short=8 master)
+    count=$(git rev-list --no-merges --count HEAD..origin/master)
+    # create a summary using the output format of "git submodule summary"
+    SUMMARY="
+* ${submodule} ${head}...${master} (${count}):
+$(git log --pretty='format:  > %s' --no-merges HEAD..origin/master)"
+
+    # fast-forward to origin/master
+    git rebase origin/master
```

Comment:
If it's a fast-forward, isn't it safer to use "git pull --ff-only" as in the original code, and not "git rebase"?